modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/Server.java

```diff
@@ -220,7 +232,7 @@ protected void initChannel(SocketChannel socketChannel) {
                          .addLast(sslHandlerFactory.create())
                          .addLast(new AmqpDecoder(amqMethodRegistryFactory.newInstance()))
                          .addLast(new AmqpEncoder())
-                         .addLast(new AmqpConnectionHandler(configuration, metricManager))
+                         .addLast(new AmqpConnectionHandler(metricManager, amqpChannelFactory))
```

Comment:
amqpChannelFactory is initialized with the metricManager, so do we need to pass the metricManager here again?

Reply:
This is done to hide AMQP channel creational dependencies from AmqpConnectionHandler.